src/prefect/orion/utilities/schemas.py

```diff
@@ -31,12 +31,16 @@ def validate(cls, v):
             raise ValueError("Unrecognized datetime.")
 
 
+B = TypeVar("B", bound=BaseModel)
+S = TypeVar("S", bound=BaseModel)
+
+
 def pydantic_subclass(
-    base: Type[BaseModel],
+    base: Type[B],
     name: str = None,
     include_fields: List[str] = None,
     exclude_fields: List[str] = None,
-) -> Type[BaseModel]:
+) -> Type[S]:
```

Comment:
This actually won't retain the type information that we care about. This is basically equivalent to what you had before. Since we're returning a _subtype_ of `B` you can just use `-> Type[B]`. It's a bit of a lie since we're modifying the fields but we do inherit from the base type still.

Reply:
Good catch, how's this version look?